runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -867,15 +900,15 @@ ClientSessionData::getClassRecord(J9Class *clazz, JITServer::ServerStream *strea
          );
          }
       }
-   else if (!record)
+   else if (uncachedClass)
       {
       // Request and cache class info from the client
       JITServerHelpers::ClassInfoTuple classInfoTuple;
       auto romClass = JITServerHelpers::getRemoteROMClass(clazz, stream, _persistentMemory, classInfoTuple);
       JITServerHelpers::cacheRemoteROMClassOrFreeIt(this, clazz, romClass, classInfoTuple);
 
       OMR::CriticalSection cs(getROMMapMonitor());
-      record = getClassRecord(clazz, missingLoaderRecord);
+      record = getClassRecord(clazz, missingLoaderInfo, uncachedClass);
```

Comment:
I think it's worth adding an assertion that the class cannot be uncached anymore.